pp.c

```diff
@@ -7215,9 +7215,11 @@ PP(pp_cmpchain_dup)
 PP(pp_isbool)
 {
     dSP;
+    dTARGET;
     SV *arg = POPs;
 
-    PUSHs(boolSV(SvIsBOOL(arg)));
+    sv_setbool(TARG, SvIsBOOL(arg));
```

Comment:
Shouldn't this be `sv_setbool_mg`?

Reply:
It should. And we should SvGETMAGIC() beforehand as well. I'll add those + some tests for it.

Reply:
Magic added in c2d575565b